### PR TITLE
make Queryable from Python UDF

### LINKED ISSUE #851: bindings for constructing Queryable from Python UDFs
Since there are no bindings for constructing a Queryable from Python, it is not possible to build interactive measurements from Python.

Simple example:

```python
def make_rr_vec_meas(prob, size):
    rr_meas = dp.m.make_randomized_response_bool(prob)
    
    def function(data):
        idx = 0
        def transition(_this, _query, _internal):
            nonlocal idx
            idx += 1
            return rr_meas(data[idx - 1])
        
        return new_queryable(transition, TO=dp.Queryable[Any, bool])
    
    return dp.c.make_user_measurement(
        dp.vector_domain(dp.atom_domain(T=bool), size=size),
        function,
        dp.symmetric_distance(),
        dp.max_divergence(T=float),
        lambda d_in: d_in * rr_meas.map(int(d_in > 0))
    )
```

### PR DESCRIPTION
**Pull request by Shoeboxam**
Closes #851

### CODE REVIEW

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #870** <a href="https://app.graphite.dev/github/pr/opendp/opendp/870" target="_blank">Graphite</a>  👈
    * **PR #871** <a href="https://app.graphite.dev/github/pr/opendp/opendp/871" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/870?utm_source=stack-comment).

File: python/test/test_interactive.py [hunk 1]
```diff
@@ -71,3 +72,27 @@ def test_sequential_composition_approxdp():
...
+        print("inside transition", query)
+        return query + 1
+    qbl = dp.c.new_user_queryable(transition, int, int)
+    print("after transition", qbl(2))
+
+    def transition(query, _is_internal):
+        print("inside transition", query)
+        return query[-1]
+    qbl = dp.c.new_user_queryable(transition, "Vec<i32>", int)
+    print("after transition", qbl([2, 3]))
```
Comment (andrewvyrros) [inline python/test/test_interactive.py:88]: Could you assert these?

File: rust/src/combinators/user_defined/ffi.rs [hunk 2]
```diff
@@ -153,3 +154,68 @@ pub extern "C" fn opendp_combinators__make_user_postprocessor(
     let _TO = TO;
     FfiResult::Ok(util::into_raw(Function::new_fallible(wrap_func(function))))
 }
+
+type TransitionFn = extern "C" fn(*const AnyObject, c_bool) -> *mut FfiResult<*mut AnyObject>;
+
+// wrap a TransitionFn in a closure, so that it can be used in Queryables
+fn wrap_trans(
```
Comment (andrewvyrros) [inline rust/src/combinators/user_defined/ffi.rs:161]: Minor: This is ambiguous with transformation, would suggest `wrap_transition`.

**Review by andrewvyrros — CHANGES_REQUESTED** [review]

**Comment by Shoeboxam** [comment]
Added 2 new commits

**Review by andrewvyrros — APPROVED** [review]
Thanks for tweaking the tests!

**Pull request merged**